### PR TITLE
allow passing in images to open chat action

### PR DESCRIPTION
**Pull request by meganrogge**
This enables extensions to add commands that run a chat request with images, for example, a screenshot that was generated.

### CODE REVIEW

File: src/vs/workbench/contrib/chat/browser/actions/chatActions.ts [hunk 1]
```diff
@@ -98,14 +99,14 @@ class OpenChatGlobalAction extends Action2 {
 		}
 		if (opts?.previousRequests?.length && chatWidget.viewModel) {
 			for (const { request, response } of opts.previousRequests) {
-				chatService.addCompleteRequest(chatWidget.viewModel.sessionId, request, undefined, 0, { message: response });
+				chatService.addCompleteRequest(chatWidget.viewModel.sessionId, request, variables ? { variables } : undefined, 0, { message: response });
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/actions/chatActions.ts:102]: Why add variables to previous requests? If you want that, shouldn't those variables be on previousRequests?

File: src/vs/workbench/contrib/chat/browser/actions/chatActions.ts [hunk 2]
```diff
@@ -98,14 +99,14 @@ class OpenChatGlobalAction extends Action2 {
...
 			}
 		}
 		if (opts?.query) {
 			if (opts.isPartialQuery) {
-				chatWidget.setInput(opts.query);
+				chatWidget.setInput(opts.query, variables);
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/actions/chatActions.ts:107]: Maybe it's easier to call `chatWidget.attachmentModel.addContext` here than add this plumbing?

File: src/vs/workbench/contrib/chat/browser/actions/chatActions.ts [hunk 3]
```diff
@@ -88,7 +89,7 @@ class OpenChatGlobalAction extends Action2 {
 		});
 	}
 
-	override async run(accessor: ServicesAccessor, opts?: string | IChatViewOpenOptions): Promise<void> {
+	override async run(accessor: ServicesAccessor, opts?: string | IChatViewOpenOptions, variables?: IChatRequestVariableEntry[]): Promise<void> {
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/actions/chatActions.ts:92]: Add it to `IChatViewOpenOptions`, but we will call it a temporary thing until screenshots are managed by vscode, `IChatRequestVariableEntry` has too much going on to be a proper argument command passed by an extension IMO.

File: src/vs/workbench/contrib/chat/browser/actions/chatActions.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/chat/browser/actions/chatActions.ts:102]: Oh, I misunderstood what was happening here.

File: src/vs/workbench/contrib/chat/browser/chatWidget.ts [hunk 4]
```diff
@@ -920,7 +920,7 @@ export class ChatWidget extends Disposable implements IChatWidget {
 				location: this.location,
 				locationData: this._location.resolveData?.(),
 				parserContext: { selectedAgent: this._lastSelectedAgent },
-				attachedContext: [...this.attachmentModel.attachments]
+				attachedContext: [...this.attachmentModel.attachments],
```
Comment (meganrogge) [inline src/vs/workbench/contrib/chat/browser/chatWidget.ts:923]:
```suggestion
				attachedContext: [...this.attachmentModel.attachments]
```

File: src/vs/workbench/contrib/chat/browser/actions/chatActions.ts [hunk 5]
```diff
@@ -55,6 +57,17 @@ export interface IChatViewOpenOptions {
 	 * Any previous chat requests and responses that should be shown in the chat view.
 	 */
 	previousRequests?: IChatViewOpenRequestEntry[];
+
+	/**
+	 * The path to the images(s)
+	 */
+	images?: IChatImageAttachment[];
```
Comment (meganrogge) [inline src/vs/workbench/contrib/chat/browser/actions/chatActions.ts:64]:
```suggestion
	/**
	 * The image(s) to include in the request
	 */
	images?: IChatImageAttachment[];
```

**Review by roblourens — APPROVED** [review]

**Pull request merged**